### PR TITLE
travis_ci: Add OSX build

### PR DESCRIPTION
**Pull request by TSC21**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,26 +1,59 @@
...
+    brew update;
+    git clone --depth 1 https://github.com/mavlink/c_library_v2.git /usr/local/include/mavlink/v2.0;
+    rm -rf /usr/local/include/mavlink/v2.0/.git;
+    fi
+
+install:
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
+    sudo -H pip install --upgrade pip setuptools;
+    sudo -H pip install nose tornado rospkg jinja2;
+    brew install eigen graphviz protobuf opencv3;
```
Comment (dagar) [inline .travis.yml]: Can we use the px4-dev and px4-sim brews here? That would be a good test for everything.
@LorenzMeier

Comment (dagar) [reply .travis.yml]: https://dev.px4.io/en/setup/dev_env_mac.html

Comment (TSC21) [reply .travis.yml]: Yes @dagar, @LorenzMeier already suggested that and it works good. I was actually testing against different Gazebo versions to check where it builds and where it fails.

File: .travis.yml [hunk 2]
```diff
@@ -1,26 +1,71 @@
...
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
+    brew update;
+    git clone --depth 1 https://github.com/mavlink/c_library_v2.git /usr/local/include/mavlink/v2.0;
+    rm -rf /usr/local/include/mavlink/v2.0/.git;
+    fi
+
+install:
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
+    sudo easy_install pip;
+    sudo -H pip install -I nose tornado rospkg Jinja2 -t /Users/travis/Library/Python/2.7/bin;
```
Comment (LorenzMeier) [inline .travis.yml]: @TSC21 Please make sure you are following the devguide instructions.

File: .travis.yml [hunk 3]
```diff
@@ -1,26 +1,71 @@
...
+    brew install eigen graphviz protobuf opencv3;
+    brew upgrade automake cmake libxml2;
+    brew link --overwrite --dry-run python@2 numpy;
```
Comment (LorenzMeier) [inline .travis.yml]: When you need to use force or overwrite flags you already know the toolchain is messed up 😈

File: .travis.yml (hunk 2 above)
Comment (TSC21) [reply .travis.yml]: I understand, but the current situation about taping `PX4/px4` is that `brew install px4-sim` will install Gazebo 8, and I wanted to have a CI build for Gazebo 9 as well.

**Comment by TSC21** [comment]
@LorenzMeier I finally managed to get the build running. The main issue was with the gazebo `homebrew` formula requiring `python@2` and `opencv` requiring `python` (Python 3) and `numpy`, which generated symlink problems (since the packages are installed under `/usr/local/Cellar`). Those are solved using `brew link --overwrite`. Then, using `pip2`, the rest of the dependencies are installed.
Note: this would also happen with `brew install px4-sim` as instructed on the devguide.

**Comment by TSC21** [comment]
Actually I just found out if I revert the order of install - `opencv` first and then `gazebo*` - I just have the need to symlink `numpy`. So, it's much cleaner now.

File: CMakeLists.txt [hunk 4]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.3 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
```
Comment (julianoes) [inline CMakeLists.txt:1]: Doesn't this trigger errors on Ubuntu 16.04?

**Review by julianoes — COMMENTED** [review]
One question, otherwise looks ok. Thanks!

File: CMakeLists.txt (hunk 4 above)
Comment (LorenzMeier) [reply CMakeLists.txt:1]: Merging this, but we can change this as needed.

**Review by LorenzMeier — APPROVED** [review]

File: CMakeLists.txt (hunk 4 above)
Comment (TSC21) [reply CMakeLists.txt:1]: it doesn't. tested on my system (16.04 with default Cmake 3.5.2).

**Comment by TSC21** [comment]
@LorenzMeier @julianoes thanks!

**Pull request merged**